pkg/controller/controller.go

```diff
@@ -233,15 +248,27 @@ func (dc *DefaultController) run(stopChan <-chan bool) {
 
 	defer close(dc.watcher)
 
+	defer dc.driver.Close()
+
+	if err := dc.driver.CreateNode("/entities", []byte{}); err != nil {
+		log.Fatalf("Unable to create overarching znode %v", err)
+	}
 	// Start a worker pool.  The pool scales up to dc.options.Workers.
 	go func() {
 		sem := semaphore.NewWeighted(int64(dc.options.Workers))
 		for watchEvent := range dc.watcher {
+			lock, canModify := dc.driver.LockEntity(watchEvent.Entity.GetName())
```

Comment:
Shouldn't we first acquire the semaphore and then lock the entity? This way we can let another controller process the entity if this controller is out of workers.

Reply:
Also, the entity name is not unique across different entity types and also across organizations. We should probably use `GetID`.